thrust/testing/complex.cu

```diff
@@ -100,227 +262,395 @@ struct TestComplexGetters
 };
 SimpleUnitTest<TestComplexGetters, FloatingPointTypes> TestComplexGettersInstance;
 
-template<typename T>
+template <typename T>
+struct TestComplexComparisionOperators
+{
+  void operator()(void)
+  {
+    {
+      thrust::host_vector<T> data = unittest::random_samples<T>(2);
+
+      const T a = data[0];
+      const T b = (data[1] == data[0]) ? data[1] + static_cast<T>(2.0) : data[1];
```

Comment:
Nitpick: it might be simpler to just create one random value and always change it
```suggestion
      thrust::host_vector<T> data = unittest::random_samples<T>(1);

      const T a = data[0];
      const T b = data[0] + static_cast<T>(2.0);
```